flink-core/src/main/java/org/apache/flink/api/common/restartstrategy/RestartStrategies.java

```diff
@@ -111,7 +118,17 @@ public static ExponentialDelayRestartStrategyConfiguration exponentialDelayResta
                 initialBackoff, maxBackoff, backoffMultiplier, resetBackoffThreshold, jitterFactor);
     }
 
-    /** Abstract configuration for restart strategies. */
+    /**
+     * Abstract configuration for restart strategies.
+     *
+     * @deprecated The {@link RestartStrategyConfiguration} class is marked as deprecated because
+     *     starting from Flink 1.19, all complex Java objects related to configuration should be
+     *     replaced by ConfigOption. In a future major version of Flink, this class will be removed
+     *     entirely. It is recommended to switch to using the ConfigOptions provided by {@link
+     *     org.apache.flink.configuration.RestartStrategyOptions} for configuring restart
+     *     strategies.
+     */
+    @Deprecated
```

Comment:
Since the whole class is marked as deprecated, do we still need to deprecated these methods?

Reply:
You're right, it's not necessary since the whole class is deprecated. I've made the changes accordingly.